### PR TITLE
Adding additional instructions around product image upload modal

### LINKED ISSUE #33559: Product creation experience: include additional guidance about product image size and format
**Description:** the product image or gallery is one of the most impactful elements of the product page. However, we don't offer any guidance on the right image size and formats to ensure that the products have compelling images.

**Acceptance criteria:**

- [ ] Update the upload modal for "Set product image" and "Add gallery images" using the [`post-upload-ui`](https://developer.wordpress.org/reference/hooks/post-upload-ui/) hook to add specific copy (TODO) about suggestion image dimensions and format, along with a link to the WooCommerce blog article about creating great product images (TODO: include link)

### PR DESCRIPTION
**Pull request by joelclimbsthings**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adding more detail around product photos appearing in the upload modal on the product page. This includes suggestion dimensions, format, and a link to a relevant blog post.

Closes #33559  .

# Screenshots

image

### How to test the changes in this Pull Request:

1. Checkout branch on fresh site
2. Go to WooCommerce -> Products and  Create Product
3. Observe the extra content in the bottom of the product image metabox.
4. Click the new link and ensure it works as expected.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33660/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Adding changelog` 9fff9c4d4d353677c629f21bd734f352a7d6e7d0
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Review by louwie17 — COMMENTED** [review]
@joelclimbsthings overall this tested well, but the `post-upload-ui` filter is used for all media modals. Meaning the copy below will also be added when users try to upload downloadable items (not image specific), or changing the featured image in a page/post.

So we probably want to limit this to product specific page and only trigger it for `Product Image` or `Product Gallery`. This latter option I couldn't find an easy way to do, maybe you know of a way?
Otherwise we could either make the copy more generic, and add something to the effect of `If uploading images: `.
Or add some JS that add's/removes this depending on the title (as the title does seem to change of the modal).

**Comment by joelclimbsthings** [comment]
Thanks @louwie17 ! Good call, hadn't fully considered that side of things. It actually seems that features images in posts/pages _don't_ actually utilize this hook, but the modal triggered for downloadable content certainly does. 

I tried to see if I could target this with CSS to only display on the proper modal but I can't seem to fashion a selector to do so. Any changes that would make a difference here would have to take place in WordPress itself, which doesn't seem worth the trouble at this point.

@pmcpinto @jarekmorawski  : 

So, in summary, we only have the option of displaying this content on _all_ upload modals shown on the Product page, including the downloadable files modal. With that in mind, do we want any changes in this copy? 

image

**Comment by pmcpinto** [comment]
@joelclimbsthings thanks for the heads up

> So, in summary, we only have the option of displaying this content on all upload modals shown on the Product page, including the downloadable files modal. With that in mind, do we want any changes in this copy?

I think we can overcome this by tweaking the copy. Something like: If you're uploading a product image, we recommend uploading JPEG files...

@jarekmorawski what do you think?

**Comment by joelclimbsthings** [comment]
Thanks @pmcpinto , I've tentatively replaced it with the following:

image

Let us know what you think @jarekmorawski

**Comment by mattsherman** [comment]
> So we probably want to limit this to product specific page and only trigger it for Product Image or Product Gallery. This latter option I couldn't find an easy way to do, maybe you know of a way?

@louwie17 @joelclimbsthings How about only calling `add_action` if on the product page? There has got to be an action hook we could use that is part of the product page lifecycle.

`current_screen` looks like it might be useful? https://developer.wordpress.org/reference/hooks/current_screen/

Or, not as ideal, inside of the `add_action`, make a check if we are on the product page?

**Comment by joelclimbsthings** [comment]
Thanks @mattsherman !
> How about only calling add_action if on the product page? There has got to be an action hook we could use that is part of the product page lifecycle.

I initially thought about doing something like this, although as it is the `post-upload-ui` only seems to be used on the downloadabe file modal, as well as the product/gallery modals, which are all already on the product page. 

We could still throw a check in there just to ensure it's not appearing elsewhere easily enough with a `get_current_screen()` or something along those lines, but I'm not sure if it's necessary.

**Comment by mattsherman** [comment]
> I initially thought about doing something like this, although as it is the `post-upload-ui` only seems to be used on the downloadabe file modal, as well as the product/gallery modals, which are all already on the product page.
> 
> We could still throw a check in there just to ensure it's not appearing elsewhere easily enough with a `get_current_screen()` or something along those lines, but I'm not sure if it's necessary.

There is nothing that guarantees this will continue to be the case in the future, or is the case when other extensions are in use, correct?

I’d err on the side of caution and performance and only call `add_action` when necessary. It seems like the more robust solution.

**Comment by jarekmorawski** [comment]
I'm afraid that adding the description like this could cause confusion. Sometimes merchants have downloadable images, which may have completely different requirements. Also, we're relying on the user to read, filter, and understand the message, rather than deliver it in a contextually meaningful way. 🤔

I wonder if it wouldn't be easier to try a different solution. Is the product image module expandable? Maybe we'd add a small bit of text inside so that it's available before people even go to the upload UI?

image

The MVP-est possible version is add a tooltip, but without the link. Even the just format and dimensions will be more helpful than complete nothing that is currently there. 😉

**Comment by joelclimbsthings** [comment]
Thanks for chiming in @jarekmorawski ! It looks like your alternative solution should be possible via the `admin_post_thumbnail_html` filter, so that's my plan. I'm at EOW so I'll plan to have this working next week. 

> There is nothing that guarantees this will continue to be the case in the future, or is the case when other extensions are in use, correct?

@mattsherman This is a great point. It looks like this shouldn't be an issue with the new filter as mentioned above, but I'll take a closer look at this next week.

**Comment by joelclimbsthings** [comment]
Thanks for the alternative @jarekmorawski , and great points overall. I have this tentatively implemented, although before the refinements (icons/etc), I wanted to mention that the metabox is much narrower than your image, so this description takes up a lot of vertical space:

image

Is that alright to move ahead with then, or do we want to think about shortening it?

**Comment by jarekmorawski** [comment]
Hi @joelclimbsthings! Thanks for the ping. The text indeed looks a bit massive. Maybe we'd cut it to:

> Upload JPEG files that are 1000 x 1000 pixels or larger (max. 2 GB). **How to prepare images?**

(91 vs 125 characters)

**Comment by joelclimbsthings** [comment]
@louwie17 I've reworked this as per the design feedback, and I believe it's ready for review again 🤞

**Review by louwie17 — APPROVED** [review]
@joelclimbsthings thanks for the changes, the final outcome looks quite nice. This tested well for brand new products or simple products.
One note is that once you save a product as Variable or Grouped, the note doesn't show up under `Product image`, we did probably want it to show there as well right?

I did approve this, as it works fine as is, and the above could be done in a separate PR.

**Comment by joelclimbsthings** [comment]
Thanks @louwie17 , good observation. After poking around I don't see an obvious way to address this concern, and my suspicion is that we're covering ~95% of the use case with this implementation. Going to call it good for now, but I'm open to revisiting in other PR.

**Comment by 3dprintingDOTagency** [comment]
How about you STOP spamming MY admin with YOUR stupid articles?

Is there a way to turn this metabox garbage off?

**Comment by simkoG** [comment]
@3dprintingDOTagency yes, you could. 

`add_filter( 'admin_post_thumbnail_html', '__return_false' );`

**Comment by oujisan2236** [comment]
Hi all is there away to remove the hint notice

Upload JPEG files that are 1000 x 1000 pixels or larger (max. 2 GB). [How to prepare images?](https://woocommerce.com/posts/fast-high-quality-product-photos/)

its repetitive as well about 8 notices underneath each other on add/edit product.

and JPEG's kinda bad file to recommend than than PNG

Time of writing this my site is in maintenance mode cause im moving servers but please see how its displayed after 6.9.1 (currently on 6.9.2)

https://www.techguysa.co.za/wp-content/uploads/2022/09/woocommerece-nonsense.png

**Comment by 3dprintingDOTagency** [comment]
add_filter( 'admin_post_thumbnail_html', '__return_false' );

That will get rid of this garbage....

> Hi all is there away to remove the hint notice
> 
> Upload JPEG files that are 1000 x 1000 pixels or larger (max. 2 GB). [How to prepare images?](https://woocommerce.com/posts/fast-high-quality-product-photos/)
> 
> its repetitive as well about 8 notices underneath each other on add/edit product.
> 
> and JPEG's kinda bad file to recommend than than PNG
> 
> Time of writing this my site is in maintenance mode cause im moving servers but please see how its displayed after 6.9.1 (currently on 6.9.2)
> 
> https://www.techguysa.co.za/wp-content/uploads/2022/09/woocommerece-nonsense.png

**Comment by 3dprintingDOTagency** [comment]
Fuck! Nevermind. That doesn't work. That removes ALL the HTML including the image itself and everything else. Some bonehead here (Gergo Simko) emailed me that code.

> add_filter( 'admin_post_thumbnail_html', '__return_false' );

**Comment by Adriandufour** [comment]
Hello, I have noticed a couple of other repositories linked and I will post an update here with any helpful updates. I wanted to check if there are any updates to this issue? I haven't seen instructions on any fixes or workarounds so I wanted to check. There was mention of creating a WooCommerce filter but I'm unfamiliar with that so if someone has a fix and can explain it that would be helpful. Thanks! I'm showing 5 instances of the instructions error on my website.

**Comment by 3dprintingDOTagency** [comment]
Sadly the Woocommerce team is great at adding useless features nobody ever asked for but not worth a shit at removing them.

**Comment by mattsherman** [comment]
@3dprintingDOTagency @oujisan2236 @Adriandufour We are working on a fix for this issue. See #34739.

**Comment by Adriandufour** [comment]
@mattsherman Thank you for the update! I will check that repository.

**Comment by mattsherman** [comment]
> @3dprintingDOTagency @oujisan2236 @Adriandufour We are working on a fix for this issue. See #34739.

This has been fixed in [WooCommerce 6.9.3](https://developer.woocommerce.com/2022/09/20/woocommerce-6-9-3-fix-release/)

**Comment by yatgirl** [comment]
I would like to chime in here as I was directed here from  [https://wordpress.org/support/topic/remove-hide-new-message-for-images-on-edit-product-edit-page/](url)

I find the space this product image upload tip takes up on every product edit page too much and unnecessary - even now that it has been corrected and reduced to only appearing once - its intrusive and is on every single product edit page - with no way to remove it.  If you wanted to make this a little tiny tooltip that brings that text up when you click on it, then ok - if you really feel the need to.  

Can I please have a code snippet or css to filter this out on the product edit page ?  I really dont want to see this on every page.

**Comment by kanlukasz** [comment]
Hello, may i ask how to remove this instruction?

At first glance `remove_action( 'current_screen', 'wc_add_product_post_thumbnail_html_filter' );` works fine.
Is this the correct way to remove this message?

**Comment by yatgirl** [comment]
@kanlukasz a code snippet that works can be found here

https://stackoverflow.com/questions/73968714/woocommerce-remove-tool-tip-on-admin-product-edit-page

**Comment by 3dprintingDOTagency** [comment]
> @kanlukasz a code snippet that works can be found here
> 
> https://stackoverflow.com/questions/73968714/woocommerce-remove-tool-tip-on-admin-product-edit-page

Thank you for helping when the FUCKING IDIOTS at Woo don't think it's a problem.
